src/root/views/Search/index.tsx

```diff
@@ -102,118 +112,158 @@ function Search(props: Props) {
       fieldReports: FieldReportTable,
     };
 
-    const scoreList = (Object.keys(resultsMap) as ResultKeys[]).map((resultKey) => ({
-      key: resultKey,
-      value: sum(resultsMap[resultKey].slice(0, MAX_VIEW_PER_SECTION).map(scoreSelector)),
-    }));
+    const scoreList = mapToList(
+      resultsMap,
+      (item, key) => ({
+        key: key as ResultKeys,
+        totalItems: item.length,
+        value: sum(item.slice(0, MAX_VIEW_PER_SECTION).map(scoreSelector))
+      }),
+    );
 
-    const sortedScoreList = scoreList.sort((a, b) => a.value - b.value);
-    const isEmpty = scoreList.every((score) => resultsMap[score.key].length === 0);
+  const keysOrdering: {
+    [key in ResultKeys]: number
+  } = {
+    regions: 0,
+    countries: 1,
+    provinces: 2,
+    emergencies: 3,
+    appeals: 3,
+    projects: 3,
+    surgeAlerts: 3,
+    surgeDeployments: 3,
+    fieldReports: 3,
+  };
+  const sortedScoreList = scoreList.sort((a, b) => (
+    keysOrdering[a.key] - keysOrdering[b.key] || b.value - a.value
+  ));
```

Comment:
This seems to be a very prone way of doing ordering. Wouldn't it be better if we pick `region`, `country` and `province` out and then render them separately?